src/data/resolvers/colony.ts

```diff
@@ -545,21 +570,103 @@ export const colonyResolvers = ({
+          colonyAddress,
+        ),
+      );
+      const extensionAddresses = await Promise.all(promises);
+      return extensionAddresses
+        .map((address: Address, idx: number) => ({
+          __typename: 'ColonyExtension',
+          colonyAddress,
+          id: address,
+          extensionId: extensions[idx],
+          address,
+        }))
+        .filter(({ address }) => address !== AddressZero);
+    },
+  },
+  ColonyExtension: {
+    async details({ address, extensionId }, { colonyAddress }) {
+      const extension = extensionData[extensionId];
+      const colonyClient = await colonyManager.getClient(
+        ClientType.ColonyClient,
         colonyAddress,
       );
-      if (extensionAddress === AddressZero) {
-        return false;
+      if (colonyClient.clientVersion === ColonyVersion.GoerliGlider) {
+        throw new Error('Colony version too old');
       }
-      return true;
+
+      const { neededColonyPermissions } = extension;
+      const roleCheckPromises = neededColonyPermissions.map(async (role) => {
+        const hasRole = await colonyClient.hasUserRole(
+          address,
+          ROOT_DOMAIN_ID,
+          role,
+        );
+        if (!hasRole) return role;
+        return null;
+      });
+
+      const roleCheckResults = await Promise.all(roleCheckPromises);
+      const missingPermissions = roleCheckResults.filter(
+        (role) => role !== null,
+      );
+
+      const installFilter = networkClient.filters.ExtensionInstalled(
+        getExtensionHash(extensionId),
+        colonyAddress,
+        null,
+      );
+      const installLogs = await getLogs(networkClient, installFilter);
+      let installedBy = AddressZero;
+      let installedAt = 0;
+
+      if (
+        installLogs[0] &&
+        installLogs[0].transactionHash &&
+        installLogs[0].blockHash
+      ) {
+        const { blockHash, transactionHash } = installLogs[0];
+        const receipt = await networkClient.provider.getTransactionReceipt(
+          transactionHash,
+        );
+        installedBy = receipt.from || AddressZero;
+        const time = await getBlockTime(networkClient.provider, blockHash);
+        installedAt = time || 0;
+      }
+
+      const extensionClient = await colonyClient.getExtensionClient(
+        extensionId,
+      );
+
+      const deprecated = await extensionClient.getDeprecated();
+
+      // If no initializationParams are present it does not need initialization
+      // and will set to be true by default
+      let initialized = !extension.initializationParams;
+      if (!initialized) {
+        // Otherwise we look for the presence of an initialization event
+        // eslint-disable-next-line max-len
+        const initializedFilter = extensionClient.filters.ExtensionInitialised();
+        const initializedEvents = await getEvents(
+          extensionClient,
+          initializedFilter,
+        );
+        initialized = !!initializedEvents.length;
```

Comment:
This value is coming from colonyJS's `getEvents` which is already wrapped in a try/catch block.

This will always return **at least** and empty array, because even if it errors out inside of the function, it falls back to an empty array.